Modules/Core/Common/src/itkRealTimeClock.cxx

```diff
@@ -35,51 +35,36 @@ RealTimeClock::RealTimeClock()
   LARGE_INTEGER frequency;
   ::QueryPerformanceFrequency(&frequency);
 
-  this->m_Frequency = static_cast<FrequencyType>(static_cast<__int64>(frequency.QuadPart));
+  m_Frequency = static_cast<FrequencyType>(frequency.QuadPart);
 
-  SYSTEMTIME st1;
-  SYSTEMTIME st2;
-  FILETIME   ft1;
-  FILETIME   ft2;
+  // Converts a FILETIME to the number of seconds since January 1, 1601.
+  const auto convertToSeconds = [](const FILETIME fileTime) {
+    const auto numberOfIntervals =
+      uint64_t{ fileTime.dwHighDateTime } * (uint64_t{ UINT32_MAX } + 1) + uint64_t{ fileTime.dwLowDateTime };
```

Comment:
Shift instead of multiplication by a power of 2?  Bitwise-or would even work.
```cpp
const auto numberOfIntervals = (uint64_t{ fileTime.dwHighDateTime } << 32) | uint64_t{ fileTime.dwLowDateTime };
```

Reply:
> Shift instead of multiplication by a power of 2? Bitwise-or would even work.

I think you're correct, thanks @Leengit. But then again, it's a matter of style, and possibly code readability. Do we prefer multiplication + addition, or bit-shift + bitwise-or?

I'm in favor of multiplication + addition in this case, but if there are more people for bit-shift + bitwise-or, I will surrender  😃

Reply:
Compiler might produce the same assembly either way. Even if it doesn't, a constructor is not a performance-critical part. I say good as-is.

Reply:
Actually my original plan was to do `const auto numberOfIntervals = bit_cast<uint64_t>(fileTime);` But then I thought it would be clearer (and safer) to use the arithmetic expression that I'm proposing here.

Reply:
I think that the result of `bit_cast` in this scenario might depend upon big endian vs. little endian; I wouldn't use it here.  As is works!

Reply:
> I think that the result of `bit_cast` in this scenario might depend upon big endian vs. little endian; I wouldn't use it here. As is works!

@Leengit OK, but the original code did memcpy from FILETIME to LARGE_INTEGER, which is basically what  `bit_cast` would do as well:
https://github.com/InsightSoftwareConsortium/ITK/blob/344e9bc58e69ed921f540c67d87dbb66fee4c466/Modules/Core/Common/src/itkRealTimeClock.cxx#L63

I was looking for a  `bit_cast` use case, by searching for "memcpy" in the code base. That's how I got here  😺

Reply:
I agree that `memcpy` and `bit_cast` are equivalent as far as endian goes.  I see at https://aakinshin.net/posts/datetime/ (search for BIGENDIAN within the page) that at least some people think endian matters in a context similar to ours.  

Though if it does matter then why haven't any of our tests failed on any system?  Regardless, the current approach is more reassuring.